src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -105,7 +107,35 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
-    const signedUrl = await this.prepareWebSocketUrl();
+    let endpointUrl = this.configuration.endpointUrl;
+    // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired
+    if (reconnecting || endpointUrl === undefined) {
+      try {
+        if (this.configuration.chimeClient.getMessagingSessionEndpoint instanceof Function) {
+          endpointUrl = (await this.configuration.chimeClient.getMessagingSessionEndpoint())
+            .Endpoint.Url;
+        } else {
+          endpointUrl = (
+            await this.configuration.chimeClient.send(new GetMessagingSessionEndpointCommand({}))
+          ).Endpoint.Url;
+        }
+        this.logger.debug(`Messaging endpoint resolved to: ${endpointUrl}`);
+      } catch (e) {
```

Comment:
nit: e -> error

Also, would be good to log the actual error as the `CloseEventHandler` will only print the custom event.